### PR TITLE
ESQL: Make Blocks and Vectors Releasable

### PR DESCRIPTION
**Pull request by ChrisHegarty**
This commit makes Blocks and Vectors Releasable.

We moving to a model where operators will release blocks that are unused as pages are processed. This allows to release things like BytesRefArrays, bigArrays, etc. As well as eventually to perform circuit breaker accountancy, when a block is no longer needed.

relates #99173

### CODE REVIEW

File: x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/data/BytesRefArrayVector.java [hunk 1]
```diff
@@ -77,4 +78,9 @@ public int hashCode() {
     public String toString() {
         return getClass().getSimpleName() + "[positions=" + getPositionCount() + ']';
     }
+
+    @Override
+    public void close() {
+        Releasables.close(values);
```
Comment (nik9000) [inline x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/data/BytesRefArrayVector.java:84]: `values.close()` is fine too. Big it super doesn't matter.

**Review by nik9000 — APPROVED** [review]
LGTM - I'd prefer to call `foo.close()` for single releasables. But it super doesn't matter.

**Review by DaveCTurner — COMMENTED** [review]
Maybe use `Releasables#closeExpectNoException` - that'll let us catch any cases where closing fails nice and early.

**Comment by nik9000** [comment]
> Maybe use `Releasables#closeExpectNoException` - that'll let us catch any cases where closing fails nice and early.

I didn't know we had that! It looks like it's designed to throw a never-caught AssertionError when we throw one of these exceptions. And we really never should throw.

**Comment by ChrisHegarty** [comment]
I updated all usage to use `closeExpectNoException`

**Review by DaveCTurner — APPROVED** [review]
Don't know much about those funky `*.st` files, but Nik seems happy :) LGTM then

**Pull request merged**